nc/source/graphics/vulkan/TechniqueManager.cpp

```diff
@@ -0,0 +1,63 @@
+#include "graphics/vulkan/TechniqueManager.h"
+#include "graphics/vulkan/Commands.h"
+#include "graphics/vulkan/techniques/SimpleTechnique.h"
+#include "component/Transform.h"
+
+namespace nc::graphics::vulkan
+{
+    TechniqueManager::TechniqueManager()
+    : m_techniques{},
+      m_globalData{}
+    {
+    }
+
+    void TechniqueManager::RegisterGlobalData(vk::Buffer* vertexBuffer, vk::Buffer* indexBuffer)
+    {
+        m_globalData.vertexBuffer = vertexBuffer;
+        m_globalData.indexBuffer = indexBuffer;
+    }
+
+    void TechniqueManager::RegisterRenderer(TechniqueType technique, Mesh mesh, Transform* transform)
+    {
+        for (auto& registeredTechnique : m_techniques)
+        {
+            if (registeredTechnique->GetType() == technique)
+            {
+                registeredTechnique->RegisterMesh(mesh);
```

Comment:
Consider making RegisterMesh/RegisterTransform a single function and the move the mesh into it.